include/filters.php

```diff
@@ -235,6 +241,25 @@ public function get_pages( $pages, $args ) {
 		return $pages;
 	}
 
+	/**
+	 * Filters the WP_Query in get_pages() per language.
+	 *
+	 * @since 3.4
+	 *
+	 * @param array $query_args  Array of arguments passed to WP_Query.
+	 * @param array $parsed_args Array of get_pages() arguments.
+	 * @return array Array of arguments passed to WP_Query with the language.
+	 */
+	public function get_pages_query_args( $query_args, $parsed_args ) {
+		if ( isset( $parsed_args['lang'] ) ) {
+			if ( $parsed_args['lang'] instanceof PLL_Language ) {
+				$parsed_args['lang'] = $parsed_args['lang']->slug;
+			}
+			$query_args['lang'] = $parsed_args['lang'];
+		}
+		return $query_args;
```

Comment:
I think we should reproduce this behavior that we have in `get_pages()`:
```php
	public function get_pages( $pages, $args ) {
		if ( isset( $args['lang'] ) && empty( $args['lang'] ) ) {
			return $pages;
		}

		$language = empty( $args['lang'] ) ? $this->curlang : $this->model->get_language( $args['lang'] );
		// ...
	}
```
Maybe like this?
```suggestion
		if ( isset( $parsed_args['lang'] ) && empty( $parsed_args['lang'] ) ) {
			return $query_args;
		}

		$language = empty( $parsed_args['lang'] ) ? $this->curlang : $this->model->get_language( $parsed_args['lang'] );

		if ( empty( $language ) ) {
			return $query_args;
		}

		$query_args['lang'] = $language->slug;

		return $query_args;
```

What da ya think?

I'm not sure about `$this->model->is_translated_post_type( $args['post_type'] )` though 🤔